index.js

```diff
@@ -154,17 +154,42 @@ export default class ClearvisioAppointmentBooker {
   }
 
   async loadStore(storeCode) {
-    var stores = await api.get(this.store, `stores?code=${storeCode}`);
-    this.store.dispatch('store/set', stores[0]);
+    try {
+      var stores = await api.get(this.store, `stores?code=${storeCode}`);
+      if ( stores[0] == undefined) {
+        this.store.dispatch('moduleState/set', 'error.storeCode');
+        return;
+      }
+      this.store.dispatch('store/set', stores[0]);
+    } catch (error) {
+      if (error.code == 403) {
+        this.store.dispatch('moduleState/set', 'error.403');
+      } else {
+        this.store.dispatch('moduleState/set', 'error.storeCode');
+      }
+    }
   }
 
   async loadEyeExaminationProcesses({eyeExaminationProcessId}) {
     if (eyeExaminationProcessId) {
-      return [await api.get(this.store, `eye_examination_processes/${eyeExaminationProcessId}`)]
-        .filter((process) => process);
+      try {
+        const  process = await api.get(this.store, `eye_examination_processes/${eyeExaminationProcessId}`);
```

Comment:
```suggestion
        return [await api.get(this.store, `eye_examination_processes/${eyeExaminationProcessId}`)];
```

legalábbis úgy hangzik számomra, h most ilyenkor 404-et kapsz tehát nem kell az erdetit "filter", tehát akkor viszont még továbbra is tömbként adjuk vissza ha eddig az volt